plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -308,6 +308,18 @@ func (r *ReferenceManager) Admit(ctx context.Context, a admission.Attributes, o
 				})
 			}
 		}
+
```

Comment:
I guess it makes sense to only perform this list + for loop if the operation is either `CREATE`, or if it is `UPDATE` **and** the namespace has changed.

Reply:
Done